### PR TITLE
chore: Make consistent forge script artifacts

### PR DESCRIPTION
**Pull request by odyslam**
## Motivation

- If some values were `None`, they would be omitted
- We need to have consistent artifacts to support #2293 

## Solution

- Change behaviour to output the artifact with a `null`
- Change `tx` to `transaction` because tx is a solidity keyword

### CODE REVIEW

**Comment by odyslam** [comment]
Relevant Question: Should we go ahead and change the `ethers` TypedTransaction fields of `type` that is also a reserved keyword (so that the solidity struct is identical to the produced JSON, or do we prefer to keep ethers intact and change the key at the solidity level?

**Review by mattsse — APPROVED** [review]
should be fine.

any concerns @joshieDo ?

**Comment by odyslam** [comment]
@mattsse what's your view on my comment above? Kinda related but also unrelated to the PR.

**Comment by mattsse** [comment]
> @mattsse what's your view on my comment above? Kinda related but also unrelated to the PR.

~~not sure I understand what the issue is wrt json parsing~~

correct me but how the fields are named is not a problem they just need to be in the correct order, right?

**Comment by odyslam** [comment]
Yeah, the issue we are tackling is whether we want the solidity structs to follow the JSON structs 1:1 for better UX.

File: cli/src/cmd/forge/script/sequence.rs [hunk 1]
```diff
@@ -241,33 +241,33 @@ pub struct TransactionWithMetadata {
     pub hash: Option<TxHash>,
     #[serde(rename = "type")]
```
Comment (joshieDo) [inline cli/src/cmd/forge/script/sequence.rs:242]: Doesn't this one need to be changed as well?

Comment (odyslam) [reply cli/src/cmd/forge/script/sequence.rs:242]: You are absolutely right, I missed this

File: testdata/script/deploy.sol [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity >=0.8.0;
+
+import {DSTest} from "../lib/ds-test/src/test.sol";
+import {Cheats} from "../cheats/Cheats.sol";
+
+contract Greeter {
```
Comment (onbjerg) [inline testdata/script/deploy.sol:7]: nit: pls format this file

**Review by onbjerg — APPROVED** [review]
lgtm

**Review by gakonst — APPROVED** [review]

**Pull request merged**